File: java/FizzBuzzOutsideIn/src/test/java/FizzBuzzTest.java
import com.mscharhag.oleaster.runner.OleasterRunner;
import org.junit.runner.RunWith;

import static com.mscharhag.oleaster.runner.StaticRunnerSupport.describe;
import static com.mscharhag.oleaster.runner.StaticRunnerSupport.it;
import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Matchers.anyInt;
import static org.mockito.Mockito.*;

@RunWith(OleasterRunner.class)
public class FizzBuzzTest {
    {
        describe("FizzBuzz", () -> {

            //Test goes here
            it("should return 1 when input is 1", () -> {
                FizzBuzz fizzBuzz = new FizzBuzz();
                String result = fizzBuzz.convert(1);
                assertThat(result).isEqualTo("1");
            });

            it("should check that all rules are invoked", () -> {
                Rule rule1 = mock(Rule.class);
                Rule rule2 = mock(Rule.class);
                FizzBuzz fizzBuzz = new FizzBuzz(rule1, rule2);
                fizzBuzz.convert(27);
                verify(rule1).appliesTo(27);
                verify(rule2).appliesTo(27);
            });

            it("should return the rule result if rule applies", () -> {
                Rule rule1 = mock(Rule.class);
                FizzBuzz fizzBuzz = new FizzBuzz(rule1);
                when(rule1.appliesTo(42)).thenReturn(true);
                when(rule1.getResult()).thenReturn("blafasel");
                String result = fizzBuzz.convert(42);
                assertThat(result).isEqualTo("blafasel");
            });
            it("should return the number if no rule applies", () -> {
                Rule rule1 = mock(Rule.class);
                FizzBuzz fizzBuzz = new FizzBuzz(rule1);
                when(rule1.appliesTo(anyInt())).thenReturn(false);

                String result = fizzBuzz.convert(4711);

                assertThat(result).isEqualTo("4711");
                verify(rule1, never()).getResult();
            });

            it("should return fizz if number is 3", () -> {

                FizzBuzz fizzBuzz = new FizzBuzz();

                String result = fizzBuzz.convert(3);

                assertThat(result).isEqualTo("Fizz");

            });


            it("should return buzz if number is 5", () -> {

                FizzBuzz fizzBuzz = new FizzBuzz();

                String result = fizzBuzz.convert(5);

                assertThat(result).isEqualTo("Buzz");

            });


            it("should return buzz-fizz if number is 15", () -> {

                FizzBuzz fizzBuzz = new FizzBuzz();

                String result = fizzBuzz.convert(15);

                assertThat(result).isEqualTo("Buzz-Fizz");

            });

        });
    }

}

File: kotlin/spring/src/test/kotlin/FizzBuzzTest.kt
import fizz.buzz.FizzBuzzInKotlin
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test

class FizzBuzzTest {
    private lateinit var service: FizzBuzzInKotlin

    @Before
    fun setUp() {
        service = FizzBuzzInKotlin()
    }

    @Test
    fun `it should calculate the number itself`() {
        assertThat(service.calculate(1)).isEqualTo("1")
        assertThat(service.calculate(2)).isEqualTo("2")
    }

    @Test
    fun `it should calculate Fizz for numbers divisible by 3`() {
        assertThat(service.calculate(3)).isEqualTo("Fizz")
        assertThat(service.calculate(6)).isEqualTo("Fizz")
    }

    @Test
    fun `it should calculate Buzz for numbers divisible by 5`() {
        assertThat(service.calculate(5)).isEqualTo("Buzz")
        assertThat(service.calculate(10)).isEqualTo("Buzz")
    }

    @Test
    fun `it should calculate Zazz for numbers divisible by 7`() {
        assertThat(service.calculate(7)).isEqualTo("Zazz")
        assertThat(service.calculate(14)).isEqualTo("Zazz")
    }

    @Test
    fun `it should calculate Fizz-Buzz-Zazz`() {
        assertThat(service.calculate(105)).isEqualTo("Fizz-Buzz-Zazz")
    }

    @Test
    fun `it should calculate up to Fizz-Buzz-Zazz`() {
        val result = service.calculateUpTo(15)
        assertThat(result).containsExactly(
                "1",
                "2",
                "Fizz",
                "4",
                "Buzz",
                "Fizz",
                "Zazz",
                "8",
                "Fizz",
                "Buzz",
                "11",
                "Fizz",
                "13",
                "Zazz",
                "Fizz-Buzz")
    }
}